api/src/Controller/Member/PostMember.php

```diff
@@ -51,53 +52,47 @@ class PostMember extends BaseMember
 
     public function buildResource(Request $request, Response $response, $args): array
     {
-        $body = $request->getParsedBody();
-        if ($body && array_key_exists('email', $body)) {
-            $body['email1'] = $body['email'];
-        }
-        if (!$body || !array_key_exists('email1', $body)) {
-            throw new InvalidParameterException("Required 'email1' parameter not present");
+        $required = ['email'];
+        $body = $this->checkRequiredBody($request, $required);
+        $good = false;
+        try {
+            $this->getMember($request, $body['email'], 'email');
+        } catch (\Exception $e) {
+            $good = true;
         }
-        $user = \lookup_users_by_email($body['email1']);
-        if (count($user['users']) > 0) {
+        if (!$good) {
             throw new ConflictException("Account with Email Already Exists");
         }
-        if (array_key_exists('legalFirstName', $body)) {
-            $body['firstName'] = $body['legalFirstName'];
-        }
-        if (array_key_exists('legalLastLast', $body)) {
-            $body['lastName'] = $body['legalLastName'];
-        }
-        if (!array_key_exists('firstName', $body) &&
-            !array_key_exists('lastName', $body)) {
-            throw new InvalidParameterException("Required 'firstName' and/or 'lastName' parameter not present");
+
+        if (!array_key_exists('legal_first_name', $body) &&
+            !array_key_exists('legal_last_name', $body)) {
+            throw new InvalidParameterException("Required 'legal_first_name' and/or 'legal_last_name' parameter not present");
         }
 
-        if (array_key_exists('Neon', $GLOBALS)) {
-            $accountID = \neon_createUser(
-                $body['email1'],
-                $body['firstName'],
-                $body['lastName']
-            );
+        $currentIdTop = Select::new($this->container->db)
```

Comment:
I assume this is here because `AccountID` is not currently a proper auto-increment primary key, but an import from Neon? I suppose as long as we're allowing for both cases, we're stuck with this. Practically speaking, it's probably safe right now, but long-term it isn't -- it isn't safe against concurrent access.

Probably we need to either A] add an auto-increment to this column or B] transition to a MemberID column that's the true primary key for the table, with the legacy AccountID being ONLY for Neon IDs. Thoughts?

Reply:
No, the reason here is that there was expressed a desire that badge number should start at `1000` and never start at just `1`.  `AccountID` is auto-incrementing but I could not find an easy way in mysql to set a minimum value..